flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/aggregations.scala

```diff
@@ -130,3 +142,63 @@ case class Avg(child: Expression) extends Aggregation {
     new SqlAvgAggFunction(AVG)
   }
 }
+
+case class UDAGGFunctionCall(
+    aggregateFunction: AggregateFunction[_, _],
+    args: Seq[Expression])
+  extends Aggregation {
+
+  override private[flink] def children: Seq[Expression] = args
+
+  // Override makeCopy method in TreeNode, to produce vargars properly
+  override def makeCopy(args: Array[AnyRef]): this.type = {
+    if (args.length < 1) {
+      throw new TableException("Invalid constructor params")
+    }
+    val agg = args.head.asInstanceOf[AggregateFunction[_, _]]
+    val arg = args.last.asInstanceOf[Seq[Expression]]
+    new UDAGGFunctionCall(agg, arg).asInstanceOf[this.type]
+  }
+
+  override def resultType: TypeInformation[_] = TypeExtractor.createTypeInfo(
+    aggregateFunction, classOf[AggregateFunction[_, _]], aggregateFunction.getClass, 0)
+
+  override def validateInput(): ValidationResult = {
+    val signature = children.map(_.resultType)
+    // look for a signature that matches the input types
+    val foundSignature = getAccumulateMethodSignature(aggregateFunction, signature)
+    if (foundSignature.isEmpty) {
+      ValidationFailure(s"Given parameters do not match any signature. \n" +
+                          s"Actual: ${signatureToString(signature)} \n" +
+                          s"Expected: ${signaturesToString(aggregateFunction, "accumulate")}")
+    } else {
+      ValidationSuccess
+    }
+  }
+
+  override def toString(): String = s"${aggregateFunction.getClass.getSimpleName}($args)"
+
+  override def toAggCall(name: String)(implicit relBuilder: RelBuilder): AggCall = {
+    val typeFactory = relBuilder.getTypeFactory.asInstanceOf[FlinkTypeFactory]
+    val sqlFunction = AggSqlFunction(name, aggregateFunction, resultType, typeFactory)
```

Comment:
I was trying to keep this name consistent with `ScalaSqlFunction` and `TableSqlFunction`